packages/server/src/integrations/postgres.ts

```diff
@@ -171,16 +166,17 @@ module PostgresModule {
       } catch (err) {
         // @ts-ignore
         throw new Error(err)
+      } finally {
+        await this.client.end()
```

Comment:
should the `buildSchema` call end as well? 

As far as I know a new instance of this class is initialized with every request, so we might create the client, set schema, connect and get the schema but not end. 

We could nearly take a similar approach to what @mike12345567 did recently with the pouch connections, where we could have:

```js
const doWithClient async (fn) {
   const client = new Client(config) 
   client.connect()
   client.query(`SET search_path TO ${this.config.schema}`)
   try {
         await fn(client)
   } finally {
      client.end()
   }
}
```

Additionally removing `this.client` to prevent any accidental misuse. Then wrap the bodies of `buildSchema` and `internalQuery` with `doWithClient`

Reply:
Good catch, didn't notice this, there is a chance of the same client being used after the `buildSchema` call - it'd probably be better if we wrapped where we used it with a connect then disconnect, rather than just connecting as part of the constructor (`setSchema` function).

Reply:
Good catch - it should. Decided not to refactor to the `doWithClient` as it's just a bit of an additional wrapper in this case and the client gets initialised as part of the `setSchema` call - so just added a `close` to `buildSchema`